inst/htmlwidgets/datatables.js

```diff
@@ -566,16 +569,17 @@ HTMLWidgets.widget({
             var val = filter.val();
             // turn off filter if in full range
             $td.data('filter', val[0] > r1 || val[1] < r2);
-            var v1 = formatDate(val[0]), v2 = formatDate(val[1]), ival;
+            var ival;
             if ($td.data('filter')) {
-              ival = v1 + ' ... ' + v2;
+              ival = formatDate(val[0], true) + ' ... ' + formatDate(val[1], true);
               $input.attr('title', ival).val(ival).trigger('input');
             } else {
               $input.attr('title', '').val('');
             }
-            $span1.text(v1); $span2.text(v2);
+            $span1.text(formatDate(val[0])); $span2.text(formatDate(val[1]));
             if (e.type === 'slide') return;  // no searching when sliding only
             if (server) {
+              ival = formatDate(val[0], true) + ' ... ' + formatDate(val[1], true);
```

Comment:
There is some redundancy in these changes from line 572 to 582. I'll revise the PR by myself to show you.